classes/Relations.php

```diff
@@ -26,61 +30,153 @@ public function relations(): Cache
     }
 
     /**
-     * @param string $class
+     * @param string $key
+     * @param string $remoteClass
+     * @return RelationBuilder
+     */
+    public function buildHasOne(string $remoteClass, string $key): RelationBuilder
+    {
+        /** @var RelationBuilder $relationBuilder */
+        $relationBuilder = $this->relations()->get(
+            $key,
+            function () use ($key, $remoteClass): RelationBuilder {
+                $localClass = get_class($this);
+                $builder    = $this->manager()->buildRemoteKey($key, $localClass, $remoteClass);
+
+                return new RelationBuilder(One::class, $builder, $this);
+            }
+        );
+
+        return $relationBuilder;
+    }
+
+    /**
+     * Returns a relation which expects the primary key of the local entity to match the foreign key of the referenced
+     * remote. The foreign key should have a unique constraint to maintain integrity.
+     *
+     * When the foreign key is not allowed to be NULL, the consuming code should make sure that when changing the
+     * remote, the old remote should be deleted or set to another value to prevent SQL exceptions upon storing.
+     *
+     * When store is called on the local entity it will be delegated to the relation. When the relation is not loaded no
+     * action is executed. However when the relation is loaded and a remote entity is available the foreign key will be
+     * set and the entity will be stored.
+     *
+     * @param string $remoteClass
      * @return One
      */
-    public function hasOne(string $class): One
+    public function hasOne(string $remoteClass): One
```

Comment:
$remoteClass instead of $class, isn't that redundant/unnecessary? I wouldn't expect something else than the remote end of the relation to be specified here.

Reply:
Improves internal consistency